Signal/src/AppDelegate.m

```diff
@@ -378,31 +378,23 @@ - (void)application:(UIApplication *)application
 }
 
 /**
- *  Signal requires an iPhone to be unlocked after reboot to be able to access keying material.
+ *  The user must unlock the device once after reboot before the database encryption key can be accessed.
  */
-- (void)verifyBackgroundBeforeKeysAvailableLaunch {
-    if ([self applicationIsActive]) {
+- (void)verifyDBKeysAvailableBeforeBackgroundLaunch
+{
+    if (UIApplication.sharedApplication.applicationState != UIApplicationStateBackground) {
         return;
     }
 
-    if (![[TSStorageManager sharedManager] databasePasswordAccessible]) {
+    if (![TSStorageManager isDatabasePasswordAccessible]) {
         UILocalNotification *notification = [[UILocalNotification alloc] init];
         notification.alertBody            = NSLocalizedString(@"PHONE_NEEDS_UNLOCK", nil);
         [[UIApplication sharedApplication] presentLocalNotificationNow:notification];
```

Comment:
I'm not sure we should present a "needs unlock" local notification. I think it'd be a better UX to say "New Messages in Signal!" or something positive like that rather than a negative or imperative message.